api/v1alpha1/module_types.go

```diff
@@ -92,42 +93,112 @@ type KernelMapping struct {
 	Regexp string `json:"regexp"`
 }
 
-// ModuleSpec describes how the OOT operator should deploy a Module on those nodes that need it.
-type ModuleSpec struct {
+type ModprobeArgs struct {
+	// Load is an optional list of arguments to be used when loading the kernel module.
+	// +kubebuilder:validation:MinItems=1
+	Load []string `json:"load,omitempty"`
+
+	// Unload is an optional list of arguments to be used when unloading the kernel module.
+	// +kubebuilder:validation:MinItems=1
+	Unload []string `json:"unload,omitempty"`
+}
 
+type ModprobeSpec struct {
+	// ModuleName is the name of the Module to be loaded.
+	ModuleName string `json:"moduleName"`
+
+	// Parameters is an optional list of kernel module parameters to be provided to modprobe.
+	Parameters []string `json:"parameters,omitempty"`
+
+	// DirName is the root directory for modules.
+	// +kubebuilder:default=/opt
+	DirName string `json:"dirName,omitempty"`
+
+	// Args is an optional list of arguments to be passed to modprobe before the name of the kernel module.
 	// +optional
-	Build *Build `json:"build"`
+	Args *ModprobeArgs `json:"args,omitempty"`
 
+	// If RawArgs are specified, they are passed straight to the modprobe binary; all other properties in this
+	// object are ignored.
 	// +optional
-	// AdditionalVolumes is a list of volumes that will be attached to the DriverContainer / DevicePlugin pod,
-	// in addition to the default ones.
-	AdditionalVolumes []v1.Volume `json:"additionalVolumes"`
+	RawArgs *ModprobeArgs `json:"rawArgs,omitempty"`
```

Comment:
Can you please show an `modprobe` command example and explain what is the difference between `Args` and `RawArgs`?

Reply:
Your typical `modprobe` command to load a module is `modprobe -v mymod`.
- `Args` lets you specify your own arguments instead of just `-v`: `modprobe ${Args} mymod`
- `RawArgs` lets you specify the entire command, in case you want to so something really custom with the arguments that KMMO does not support: `modprobe ${RawArgs}`

Specify `RawArgs` makes KMMO skip the generation of the CLI - it just appends whatever you provided to `modprobe` and runs that.

Reply:
Thanks. It make more sense know.

I do think we should add those examples you put in this comment inside the fields description for clarification though.

Reply:
Here are the current comments:
```go
// Args is an optional list of arguments to be passed to modprobe before the name of the kernel module.

// If RawArgs are specified, they are passed straight to the modprobe binary; all other properties in this
// object are ignored.
```
Do you think they are not clear enough?

Reply:
I think the examples are super valuable here, I have commented on each one of them individually.
Resolving this thread.